static/app/components/menuListItem.tsx

```diff
@@ -0,0 +1,255 @@
+import {forwardRef as reactForwardRef} from 'react';
+import isPropValid from '@emotion/is-prop-valid';
+import styled from '@emotion/styled';
+
+import overflowEllipsis from 'sentry/styles/overflowEllipsis';
+import space from 'sentry/styles/space';
+
+/**
+ * Menu item priority. Currently there's only one option other than default,
+ * but we may choose to add more in the future.
+ */
+type Priority = 'danger' | 'default';
+
+export type MenuListItemProps = {
+  /**
+   * Optional descriptive text. Like 'label', should preferably be a string or
+   * have appropriate aria-labels.
+   */
+  details?: React.ReactNode;
+  /**
+   * Item label. Should prefereably be a string. If not, make sure that
+   * there are appropriate aria-labels.
+   */
+  label?: React.ReactNode;
+  /*
+   * Items to be added to the left of the label
+   */
+  leadingItems?: React.ReactNode;
+  /*
+   * Whether leading items should be centered with respect to the entire
+   * height of the item. If false (default), they will be centered with
+   * respect to the first line of the label element.
+   */
+  leadingItemsSpanFullHeight?: boolean;
+  /**
+   * Accented text and background (on hover) colors.
+   */
+  priority?: Priority;
+  /**
+   * Whether to show a line divider below this item
+   */
+  showDivider?: boolean;
+  /*
+   * Items to be added to the right of the label.
+   */
+  trailingItems?: React.ReactNode;
+  /*
+   * Whether trailing items should be centered wrt/ the entire height of the
+   * item. If false (default), they will be centered wrt/ the first line of
+   * the label element.
+   */
+  trailingItemsSpanFullHeight?: boolean;
+};
+
+type OtherProps = {
+  as?: React.ElementType;
```

Comment:
I would recommend against using the `styled` as thing if possible. Looks like we were already using it before though damn